packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -388,99 +390,102 @@ export function createElement(
+    const firstChild = ((div.firstChild: any): HTMLScriptElement);
+    domElement = div.removeChild(firstChild);
+  } else if (typeof props.is === 'string') {
+    domElement = ownerDocument.createElement(type, {is: props.is});
+  } else {
+    // Separate else branch instead of using `props.is || undefined` above because of a Firefox bug.
+    // See discussion in https://github.com/facebook/react/pull/6896
+    // and discussion in https://bugzilla.mozilla.org/show_bug.cgi?id=1276240
+    domElement = ownerDocument.createElement(type);
+    // Normally attributes are assigned in `setInitialDOMProperties`, however the `multiple` and `size`
+    // attributes on `select`s needs to be added before `option`s are inserted.
+    // This prevents:
+    // - a bug where the `select` does not scroll to the correct option because singular
+    //  `select` elements automatically pick the first item #13222
+    // - a bug where the `select` set the first item as selected despite the `size` attribute #14239
+    // See https://github.com/facebook/react/issues/13222
+    // and https://github.com/facebook/react/issues/14239
+    if (type === 'select') {
+      const node = ((domElement: any): HTMLSelectElement);
+      if (props.multiple) {
+        node.multiple = true;
+      } else if (props.size) {
+        // Setting a size greater than 1 causes a select to behave like `multiple=true`, where
+        // it is possible that no option is selected.
+        //
+        // This is only necessary when a select in "single selection mode".
+        node.size = props.size;
       }
     }
-  } else {
-    domElement = ownerDocument.createElementNS(namespaceURI, type);
   }
 
   if (__DEV__) {
-    if (namespaceURI === HTML_NAMESPACE) {
-      if (
-        !isCustomComponentTag &&
-        // $FlowFixMe[method-unbinding]
-        Object.prototype.toString.call(domElement) ===
-          '[object HTMLUnknownElement]' &&
-        !hasOwnProperty.call(warnedUnknownTags, type)
-      ) {
-        warnedUnknownTags[type] = true;
-        console.error(
-          'The tag <%s> is unrecognized in this browser. ' +
-            'If you meant to render a React component, start its name with ' +
-            'an uppercase letter.',
-          type,
-        );
-      }
+    if (
+      !isCustomComponentTag &&
+      // $FlowFixMe[method-unbinding]
+      Object.prototype.toString.call(domElement) ===
+        '[object HTMLUnknownElement]' &&
+      !hasOwnProperty.call(warnedUnknownTags, type)
+    ) {
+      warnedUnknownTags[type] = true;
+      console.error(
+        'The tag <%s> is unrecognized in this browser. ' +
+          'If you meant to render a React component, start its name with ' +
+          'an uppercase letter.',
+        type,
+      );
     }
   }
 
   return domElement;
 }
 
+// Creates elements in the HTML either SVG or Math namespace
+export function createElementNS(
+  namespaceURI: ExoticNamespace,
```

Comment:
responsbility is now on the caller to provide the namespace for this instance rather than the parent namespace